onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -29,9 +30,19 @@
 
 
 def replace_special_characters_with_underscores(data):
+    # Remove metadata fields from the column headers
+    remove_metadata_fields(data)
     return [re.sub(r"\W", r"_", a) for a in data]
 
 
+def remove_metadata_fields(data):
+    for field in METADATA_FIELDS:
+        if field in data:
+            data.remove(field)
+        list(set(data))
+    return data
```

Comment:
Is it possible to have this functionality within the Data serializer ? I'm guessing that's where it's used ?

You can do so by using the [`exclude` variable](https://www.django-rest-framework.org/api-guide/serializers/#specifying-which-fields-to-include)